src/plugins.star

```diff
@@ -0,0 +1,78 @@
+    "AutoGPTDiscord": {"repository": "gravelBridge/AutoGPT-Discord", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: ["DISCORD_BOT_TOKEN", "AUTHORIZED_USER_IDS", "BOT_PREFIX", "CHANNEL_ID", "ASK_FOR_INPUT"]}, # TODO ASK_FOR_INPUT=True can be auto set
+    "AutoGPTDollyPlugin": {"repository": "pr-0f3t/Auto-GPT-Dolly-Plugin", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: []},
+    "AutoGPTGoogleAnalyticsPlugin": {"repository": "isaiahbjork/Auto-GPT-Google-Analytics-Plugin", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["GOOGLE_APPLICATION_CREDENTIALS", "GOOGLE_ANALYTICS_VIEW_ID"]}, # TODO GOOGLE_APPLICATION_CREDENTIALS=firebase.json can be auto set
+    "AutoGPT_IFTTT": {"repository": "AntonioCiolino/AutoGPT-IFTTT", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["IFTTT_WEBHOOK_TRIGGER_NAME", "IFTTT_KEY"]},
+    "AutoGPT_YouTube": {"repository": "jpetzke/AutoGPT-YouTube", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["YOUTUBE_API_KEY"]},
+    "AutoGPTPMPlugin": {"repository": "minfenglu/AutoGPT-PM-Plugin", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: ["TRELLO_API_KEY", "TRELLO_API_TOKEN", "TRELLO_CONFIG_FILE"]},
+    "AutoGPTWolframAlpha": {"repository":"gravelBridge/AutoGPT-WolframAlpha", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: ["WOLFRAM_ALPHA_APP_ID"]},
+    "AutoGPTTodoistPlugin": {"repository": "danikhan632/Auto-GPT-Todoist-Plugin", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["TODOIST_TOKEN"]},
+    "AutoGPTMessagesPlugin": {"repository": "danikhan632/Auto-GPT-Messages-Plugin", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["IMESSAGE_PASSWORD_KEY", "IMESSAGE_BASE_URL"]},
+    "AutoGPTWebInteraction": {"repository": "gravelBridge/AutoGPT-Web-Interaction", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: []},
+    "AutoGPTNotion": {"repository": "doutv/Auto-GPT-Notion", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: ["NOTION_TOKEN", "NOTION_DATABASE_ID"]},
+    "SystemInformationPlugin": {"repository": "hdkiller/Auto-GPT-SystemInfo", "branch": MASTER_BRANCH, REQUIRED_ENV_VARS_KEY: []},
+    "AutoGPT_Zapier": {"repository": "AntonioCiolino/AutoGPT-Zapier", "branch": MAIN_BRANCH, REQUIRED_ENV_VARS_KEY: ["ZAPIER_WEBHOOK_ENDPOINT"]},
+}
+
+def get_plugin_url(plugin_data, plugin_branch_to_use, plugin_repo_to_use):
+    repo = plugin_data["repository"]
+    if plugin_repo_to_use:
+        repo = plugin_repo_to_use
+    branch = plugin_data["branch"]
+    if plugin_branch_to_use:
+        branch = plugin_branch_to_use
+    return "https://github.com/{0}/archive/refs/heads/{1}.zip".format(repo, branch)
+
+def get_filename(plugin):
+    author, actual_repo = plugin["repository"].split("/")
+    return actual_repo + ZIP_EXTENSION
+
+def areAllRequiredEnvVarsSet(env_vars, plugins_names):
+    areAllRequiredEnvVarsSet = True
+    missing_env_vars = []
```

Comment:
can this be a map that points to a list

so the results are like

missing_env_vars["AutoGPTTWitter"] = ["TwitterAPiKey"]
missing_env_vars["AutoGPTEmailPlugin"] = ["EMAIL_ADDRESS"]

so that when we fail we can print which plugin needed what?